mongorestore/restore.go

```diff
@@ -25,24 +25,25 @@ import (
 
 const insertBufferFactor = 16
 
-var indexOptions = map[string]bool{
-	"bits": true,
-	"min":  true,
-	"max":  true,
-	"coarsestIndexedLevel": true,
-	"finestIndexedLevel":   true,
-	"2dsphereIndexVersion": true,
-	"background":           true,
-	"collation":            true,
-	"default_language":     true,
-	"dropDups":             true,
-	"expireAfterSeconds":   true,
-	"bucketSize":           true,
-	"name":                 true,
-	"v":                    true,
-	"key":                  true,
-	"language_override":    true,
-	"ns":                   true,
+// validIndexOptions are taken from https://github.com/mongodb/mongo/blob/master/src/mongo/db/index/index_descriptor.h
+var validIndexOptions = map[string]bool{
```

Comment:
nit: I suggest sorting the list of options below so the order isn't so random.  (Keeps readers from wondering if there is any meaning to the order.)